### PR TITLE
[STORM-1112] add executor-id to executor thread name

### PR DESCRIPTION
**Pull request by zhuoliu**

### CODE REVIEW

File: storm-core/src/clj/backtype/storm/daemon/executor.clj [hunk 1]
```diff
@@ -685,7 +685,7 @@
           0))
       :kill-fn (:report-error-and-die executor-data)
       :factory? true
-      :thread-name component-id)]))
+      :thread-name (str component-id "-executor" (:executor-id executor-data)))]))
```
Comment (HeartSaVioR) [inline storm-core/src/clj/backtype/storm/daemon/executor.clj]: @zhuoliu 
nit: I'd like to add a "-" between "-executor" and executor-id, but it's related to personal style, and really minor.

**Comment by HeartSaVioR** [comment]
@zhuoliu 
Great work, +1

File: storm-core/src/clj/backtype/storm/daemon/executor.clj (hunk 1 above)
Comment (zhuoliu) [inline storm-core/src/clj/backtype/storm/daemon/executor.clj]: Thanks @HeartSaVioR 
Currently the executor send queue thread has name like this:
 "Thread-25-disruptor-executor[25 25]-send-queue"

I take similar format to above to be consistent, now we will have executor thread name as:
"Thread-26-spout-executor[25 25]"

Comment (HeartSaVioR) [inline storm-core/src/clj/backtype/storm/daemon/executor.clj]: @zhuoliu Ah, OK. I prefer current formatting. :)

**Comment by revans2** [comment]
+1

**Pull request merged**